app/models/concerns/user_subscription_sourceable.rb

```diff
@@ -4,7 +4,7 @@ module UserSubscriptionSourceable
   # This all assumes there's an association with User under the column user_id.
 
   included do
-    has_many :user_subscriptions, as: :user_subscription_sourceable
+    has_many :user_subscriptions, as: :user_subscription_sourceable, dependent: :destroy
```

Comment:
Destroying the object of a subscription, means also destroying the subscriptions attached to it. If the author removes an articles, subscription should go as well

Reply:
Great catch! I don't think we would want to add this `dependent: :destroy`.

The subscriptions should remain even if an author destroyed one of their articles. We would end up displaying it as a subscription to an unpublished/deleted article.  Since it's the author's content, that's sort of on them.

I'm also thinking of how this could look in the future. If we set aside what we know about the liquid tag currently using this, we can use subscriptions in other areas. We could add a button to a user's profile where another user could click "subscribe", for example (also set aside the fact that would be kind of confusing what the difference between subscribing to and following a user is). These subscriptions are logging the bridge between 2 users and where the bridge came from originally. Even if the origin (Article, Comment, User profile, etc) were to be deleted, the subscription itself is still relevant.

Technically, we could do subscriptions with no source at all which is what it ultimately becomes if the source is deleted. It's less useful to the author because they can't segment their audience. In the context of deleting the source though, the source was once present before the author chose to delete their content, so they can still segment and use that power before they delete it.

We also "soft delete" many objects too which helps with this.

Reply:
@atsmith813 thanks for the explanation. I'm going to make the following changes then:

- set this as `dependent: nullify` so that they are automatically set to null if the source gets destroyed
- set `belongs_to :user_subscription_sourceable, polymorphic: true` in the `UserSubscription` model to `optional: true` as the `UserSubscription` can be valid without it as well. 

Would that work?

Reply:
@rhymes that would work, I like that! I wonder if there's a quick win to validate the presence of `user_subscription_sourceable` only on create 🤔 . A subscription shouldn't be created without it, but it can be updated to be removed.

Thinking out loud...something we could explore in the future (definitely not now because it doesn't even exist yet), is somehow persisting the `user_subscription_sourceable_type` value so we could then say display _Deleted Comment_, _Deleted Article_, etc. Since we only use this in Articles and we don't even have a use case/view for this scenario anyways, we can re-visit the idea if/when we expand subscriptions.

What do you think?